poly_commit/src/hyrax/inner_prod_argument.rs

```diff
@@ -0,0 +1,213 @@
+    pub com_ip_ry: C,
+    pub sigma_3rd_masked_r_mu_x: Vec<C::Scalar>,
+    pub sigma_3rd_masked_r_mu_x_randomness: C::Scalar,
+    pub sigma_3rd_masked_eval_randomness: C::Scalar,
+}
+
+impl<C: CurveAffine + FieldSerde> FieldSerde for PedersenIPAProof<C>
+where
+    C::Scalar: FieldSerde,
+{
+    const SERIALIZED_SIZE: usize = unimplemented!();
+
+    fn serialize_into<W: std::io::Write>(&self, mut writer: W) -> arith::FieldSerdeResult<()> {
+        self.com_r.serialize_into(&mut writer)?;
+        self.com_zero.serialize_into(&mut writer)?;
+        self.com_ip_ry.serialize_into(&mut writer)?;
+        self.sigma_3rd_masked_r_mu_x.serialize_into(&mut writer)?;
+        self.sigma_3rd_masked_r_mu_x_randomness
+            .serialize_into(&mut writer)?;
+        self.sigma_3rd_masked_eval_randomness
+            .serialize_into(&mut writer)
+    }
+
+    fn deserialize_from<R: std::io::Read>(mut reader: R) -> arith::FieldSerdeResult<Self> {
+        let com_r: C = C::deserialize_from(&mut reader)?;
+        let com_zero: C = C::deserialize_from(&mut reader)?;
+        let com_ip_ry: C = C::deserialize_from(&mut reader)?;
+        let sigma_3rd_masked_r_mu_x: Vec<C::Scalar> = Vec::deserialize_from(&mut reader)?;
+        let sigma_3rd_masked_r_mu_x_randomness = C::Scalar::deserialize_from(&mut reader)?;
+        let sigma_3rd_masked_eval_randomness = C::Scalar::deserialize_from(&mut reader)?;
+        Ok(Self {
+            com_r,
+            com_zero,
+            com_ip_ry,
+            sigma_3rd_masked_r_mu_x,
+            sigma_3rd_masked_r_mu_x_randomness,
+            sigma_3rd_masked_eval_randomness,
+        })
+    }
+}
+
+pub(crate) fn pedersen_ipa_prove<C, T>(
+    params: &PedersenParams<C>,
+    x_vec: &[C::Scalar],
+    y_vec: &[C::Scalar],
+    com_randomness: C::Scalar,
+    transcript: &mut T,
+) -> PedersenIPAProof<C>
+where
+    C: CurveAffine + FieldSerde,
+    T: Transcript<C::Scalar>,
+    C::Scalar: ExtensionField + PrimeField,
+    C::ScalarExt: ExtensionField + PrimeField,
+{
+    let mut pedersen_commit_ro_absorb = |gs: &[C::Scalar]| -> (C, C::Scalar) {
+        let (com, r_mask) = pedersen_commit(params, gs);
+
+        let mut com_bytes: Vec<u8> = Vec::new();
+        com.serialize_into(&mut com_bytes).unwrap();
+        transcript.append_u8_slice(&com_bytes);
+        (com, r_mask)
+    };
+
+    // NOTE(HS) we want some randomness in sigma protocol masking, but not from transcript.
+    let r_vec: Vec<C::Scalar> = {
+        let mut rng = thread_rng();
+        (0..x_vec.len())
+            .map(|_| C::Scalar::random(&mut rng))
+            .collect()
+    };
+    let (com_r, rs_mask) = pedersen_commit_ro_absorb(&r_vec);
+
+    let (com_zero, zero_mask) = pedersen_commit_ro_absorb(&[]);
+
+    let ip_ry: C::Scalar = izip!(&r_vec, y_vec).map(|(r, y)| *r * *y).sum();
+    let (com_ip_ry, ip_mask) = pedersen_commit_ro_absorb(&[ip_ry]);
+
+    let mu: C::Scalar = transcript.generate_challenge_field_element();
```

Comment:
the thing surprises me is that - in jolt impl there is actually no IPA, they sent over the folded row of multilinear poly, and verify runs eval check and runs pedersen commitment to see if it matches the linear combination of prior commitments - supposing we are not looking for zero knowledge property here, I suppose we can just ditch the IPA here and just return the top var fixed multilinear poly, which save a bit on randomness and perf?

Reply:
yea removing whole IPA and sigma protocol completely in https://github.com/PolyhedraZK/Expander/pull/200/commits/57d3a1559bc78c148fb18b9fdd6986fb768ba67d for no simulatability